src/python/WMCore/WMStats/CherryPyThreads/BuildParentLock.py

```diff
@@ -52,13 +59,15 @@ def fetchIncludeParentsRequests(self, config):
                 setParents.add(res[0]['parent_dataset'])
 
         if not incompleteParentage:
-            DataCache.setParentDatasetList(list(setParents))
-            self.logger.info("Parentage lookup complete and cache renewed")
+            dictParents['parentlocks'] = list(setParents)
+            self.reqmgrAux.updateParentLocks(dictParents)
+            self.logger.info("Parentage lookup complete and auxiliary database updated")
         else:
             # then don't replace any data for the moment, simply add new parents
             previousData = set(DataCache.getParentDatasetList())
```

Comment:
This should actually be something like `self.reqmgrAux.getParentLocks()`, thus this client get method needs to be implemented as well.
DataCache object will only be used to filter out workflows requiring IncludeParents, nothing else.

Reply:
Oof, yeah. Missed that line completely.